### ISSUE TITLE
Impossibility to sign-in when RandomKeyPassword missing

### ISSUE THREAD

**Issue by LeJ84** [aws-amplify/amplify-js#9366]
### JavaScript Framework

Not applicable

### Amplify APIs

Authentication

### Amplify Categories

auth

### Environment information

<details>

```
# Put output below this line
System:
    OS: Linux 5.13 Pop!_OS 21.04
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
    Memory: 429.96 MB / 15.36 GB
    Container: Yes
    Shell: 5.1.4 - /bin/bash
  Binaries:
    Node: 14.17.6 - /usr/bin/node
    npm: 6.14.11 - ~/.local/bin/npm
  Browsers:
    Chrome: 94.0.4606.54
    Firefox: 94.0
  npmPackages:
    @aws-amplify/ui-react: ^1.2.2 => 1.2.4 
    @testing-library/jest-dom: ^5.14.1 => 5.14.1 
    @testing-library/react: ^11.2.7 => 11.2.7 
    @testing-library/user-event: ^12.8.3 => 12.8.3 
    aws-amplify: ^4.1.0 => 4.1.2 
    buffer: ^6.0.3 => 6.0.3 (4.9.2, 5.7.1)
    classnames: ^2.3.1 => 2.3.1 
    i18next: ^20.3.2 => 20.3.2 
    i18next-browser-languagedetector: ^6.1.2 => 6.1.2 
    jsonwebtoken: ^8.5.1 => 8.5.1 
    node-sass: ^6.0.0 => 6.0.1 
    react: ^17.0.2 => 17.0.2 
    react-dom: ^17.0.2 => 17.0.2 
    react-i18next: ^11.11.0 => 11.11.0 
    react-router-dom: ^5.2.0 => 5.2.0 
    react-scripts: 4.0.3 => 4.0.3 
    web-vitals: ^1.1.2 => 1.1.2 
  npmGlobalPackages:
    @aws-amplify/cli: 7.6.2
    amplify: 0.0.11
    geckodriver: 2.0.2
    npm: 6.14.11
    selenium-side-runner: 3.17.0
    serve: 12.0.0

```

</details>

### Describe the bug

Impossibility to sign in using amplify authenticator...Sign in method throwing and error.

### Expected behavior

When i sign-in with correct information to be logged without problem

### Reproduction steps

1 - Log in your application with amplify authenticator
2 - Logout
3 - Delete only the RamdomPasswordKey from the localStorage
4 - You can't sign-in anymore   

### Additional information and screenshots

This case looks pretty weird and rare but quite annoying to encounter. Perhaps, some little condition can fix it without too much trouble.

**Comment by chrisbonifacio** [comment]
Hi @LeJ84 👋 thank you for raising this issue. We've since upgraded the UI components. Would you like to try using the new authenticator component?

You can upgrade your UI package by running `npm i @aws-amplify/ui-react@latest`

**Comment by LeJ84** [comment]
I've made the update but it don't seems to fix that problem but i don't use the authenticator component because of a specific flow. I've made custom components calling sign.In, signUp, signOut ... methods. With the authenticator you don't have that issue ?

**Comment by chrisbonifacio** [comment]
I don't see a key being set to localStorage called randomPasswordKey. Can you share screenshots or a loom video of the behavior, with localStorage open?

**Comment by chrisbonifacio** [comment]
Do you have remember device enabled? Maybe that's why I don't see it, it seems like it's part of a device login process.

**Comment by LeJ84** [comment]
Yes remember device is enabled and set to always !

**Comment by chrisbonifacio** [comment]
Hi @LeJ84 👋 sorry for the delay. So, I was able to reproduce this issue consistently. I'll label it as a bug for the team to investigate further.

From what I could tell, if you remove the `deviceKey` as well, you should be able to login again.

**Comment by AllanZhengYP** [comment]
Hi @LeJ84 We just release the fix in the version `aws-amplify@6.1.4`. Can you check if that version fixes the issue?
